### ISSUE TITLE
你好，我想问一下，我这训练是什么情况？

### ISSUE THREAD

**Issue by louielu1027** [guoyilin/caffe#3]
I0811 04:43:50.113242  6500 solver.cpp:214] Iteration 0, loss = 0
I0811 04:43:50.113284  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:43:50.113313  6500 solver.cpp:486] Iteration 0, lr = 0.0001
I0811 04:44:57.945654  6500 solver.cpp:214] Iteration 18, loss = 48.8384
I0811 04:44:57.945754  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:44:57.945766  6500 solver.cpp:486] Iteration 18, lr = 0.0001
I0811 04:46:05.653295  6500 solver.cpp:214] Iteration 36, loss = 16.5096
I0811 04:46:05.653362  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:46:05.653375  6500 solver.cpp:486] Iteration 36, lr = 0.0001
I0811 04:47:13.365404  6500 solver.cpp:214] Iteration 54, loss = -3.05176e-05
I0811 04:47:13.365469  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:47:13.365480  6500 solver.cpp:486] Iteration 54, lr = 0.0001
I0811 04:48:21.204015  6500 solver.cpp:214] Iteration 72, loss = 0.703822
I0811 04:48:21.204080  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:48:21.204092  6500 solver.cpp:486] Iteration 72, lr = 0.0001
I0811 04:49:29.063345  6500 solver.cpp:214] Iteration 90, loss = 0.234567
I0811 04:49:29.063438  6500 solver.cpp:229]     Train net output #0: tripletloss = 2.11134 (\* 1 = 2.11134 loss)
I0811 04:49:29.063448  6500 solver.cpp:486] Iteration 90, lr = 0.0001
I0811 04:50:36.787369  6500 solver.cpp:214] Iteration 108, loss = 1.7714
I0811 04:50:36.787441  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:50:36.787454  6500 solver.cpp:486] Iteration 108, lr = 0.0001
I0811 04:51:44.535688  6500 solver.cpp:214] Iteration 126, loss = -6.11544e-05
I0811 04:51:44.535756  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:51:44.535769  6500 solver.cpp:486] Iteration 126, lr = 0.0001
I0811 04:52:52.248569  6500 solver.cpp:214] Iteration 144, loss = 12.0253
I0811 04:52:52.248634  6500 solver.cpp:229]     Train net output #0: tripletloss = 14.9139 (\* 1 = 14.9139 loss)
I0811 04:52:52.248643  6500 solver.cpp:486] Iteration 144, lr = 0.0001
I0811 04:53:59.917649  6500 solver.cpp:214] Iteration 162, loss = -6.27041e-05
I0811 04:53:59.917714  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:53:59.917728  6500 solver.cpp:486] Iteration 162, lr = 0.0001
I0811 04:55:07.621796  6500 solver.cpp:214] Iteration 180, loss = 0.0672214
I0811 04:55:07.621862  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:55:07.621872  6500 solver.cpp:486] Iteration 180, lr = 0.0001
I0811 04:56:15.294675  6500 solver.cpp:214] Iteration 198, loss = 2.1794
I0811 04:56:15.294742  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:56:15.294755  6500 solver.cpp:486] Iteration 198, lr = 0.0001
I0811 04:57:22.965425  6500 solver.cpp:214] Iteration 216, loss = 0.459415
I0811 04:57:22.965522  6500 solver.cpp:229]     Train net output #0: tripletloss = 4.13529 (\* 1 = 4.13529 loss)
I0811 04:57:22.965535  6500 solver.cpp:486] Iteration 216, lr = 0.0001
I0811 04:58:31.153684  6500 solver.cpp:214] Iteration 234, loss = 0.0995185
I0811 04:58:31.153771  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:58:31.153781  6500 solver.cpp:486] Iteration 234, lr = 0.0001
I0811 04:59:39.472342  6500 solver.cpp:214] Iteration 252, loss = 1.246
I0811 04:59:39.472421  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 04:59:39.472436  6500 solver.cpp:486] Iteration 252, lr = 0.0001
I0811 05:00:47.818704  6500 solver.cpp:214] Iteration 270, loss = 2.32974
I0811 05:00:47.818778  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:00:47.818792  6500 solver.cpp:486] Iteration 270, lr = 0.0001
I0811 05:01:56.114686  6500 solver.cpp:214] Iteration 288, loss = 0.0435698
I0811 05:01:56.114778  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:01:56.114791  6500 solver.cpp:486] Iteration 288, lr = 0.0001
I0811 05:03:04.489054  6500 solver.cpp:214] Iteration 306, loss = 2.23856
I0811 05:03:04.489162  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:03:04.489187  6500 solver.cpp:486] Iteration 306, lr = 0.0001
I0811 05:04:12.155246  6500 solver.cpp:214] Iteration 324, loss = 101.816
I0811 05:04:12.155308  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:04:12.155318  6500 solver.cpp:486] Iteration 324, lr = 0.0001
I0811 05:05:19.840358  6500 solver.cpp:214] Iteration 342, loss = 0.825645
I0811 05:05:19.840417  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:05:19.840427  6500 solver.cpp:486] Iteration 342, lr = 0.0001
I0811 05:06:27.506650  6500 solver.cpp:214] Iteration 360, loss = 0.746133
I0811 05:06:27.506710  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:06:27.506718  6500 solver.cpp:486] Iteration 360, lr = 0.0001
I0811 05:07:35.187403  6500 solver.cpp:214] Iteration 378, loss = -5.87702e-05
I0811 05:07:35.187466  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:07:35.187479  6500 solver.cpp:486] Iteration 378, lr = 0.0001
I0811 05:08:43.087249  6500 solver.cpp:214] Iteration 396, loss = 0.414626
I0811 05:08:43.087309  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:08:43.087319  6500 solver.cpp:486] Iteration 396, lr = 0.0001
I0811 05:09:50.796950  6500 solver.cpp:214] Iteration 414, loss = 2.79512
I0811 05:09:50.797013  6500 solver.cpp:229]     Train net output #0: tripletloss = 10.4305 (\* 1 = 10.4305 loss)
I0811 05:09:50.797021  6500 solver.cpp:486] Iteration 414, lr = 0.0001
I0811 05:10:58.498670  6500 solver.cpp:214] Iteration 432, loss = 3.24082
I0811 05:10:58.498751  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:10:58.498765  6500 solver.cpp:486] Iteration 432, lr = 0.0001
I0811 05:12:06.169255  6500 solver.cpp:214] Iteration 450, loss = 1.53988
I0811 05:12:06.169314  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:12:06.169327  6500 solver.cpp:486] Iteration 450, lr = 0.0001
I0811 05:13:13.850697  6500 solver.cpp:214] Iteration 468, loss = 1.0264
I0811 05:13:13.850757  6500 solver.cpp:229]     Train net output #0: tripletloss = 9.23815 (\* 1 = 9.23815 loss)
I0811 05:13:13.850766  6500 solver.cpp:486] Iteration 468, lr = 0.0001
I0811 05:14:21.507980  6500 solver.cpp:214] Iteration 486, loss = 1.96649
I0811 05:14:21.508039  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:14:21.508051  6500 solver.cpp:486] Iteration 486, lr = 0.0001
I0811 05:15:29.194263  6500 solver.cpp:214] Iteration 504, loss = -5.94854e-05
I0811 05:15:29.194319  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:15:29.194329  6500 solver.cpp:486] Iteration 504, lr = 0.0001
I0811 05:16:36.892102  6500 solver.cpp:214] Iteration 522, loss = 1.02462
I0811 05:16:36.892195  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:16:36.892205  6500 solver.cpp:486] Iteration 522, lr = 0.0001
I0811 05:17:44.602576  6500 solver.cpp:214] Iteration 540, loss = -5.96487e-05
I0811 05:17:44.602639  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:17:44.602649  6500 solver.cpp:486] Iteration 540, lr = 0.0001
I0811 05:18:52.329092  6500 solver.cpp:214] Iteration 558, loss = 0.168909
I0811 05:18:52.329149  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:18:52.329159  6500 solver.cpp:486] Iteration 558, lr = 0.0001
I0811 05:20:00.048117  6500 solver.cpp:214] Iteration 576, loss = -6.00815e-05
I0811 05:20:00.048174  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:20:00.048185  6500 solver.cpp:486] Iteration 576, lr = 0.0001
I0811 05:21:07.706097  6500 solver.cpp:214] Iteration 594, loss = -6.01411e-05
I0811 05:21:07.706153  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:21:07.706162  6500 solver.cpp:486] Iteration 594, lr = 0.0001
I0811 05:22:15.423184  6500 solver.cpp:214] Iteration 612, loss = 0.130687
I0811 05:22:15.423243  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:22:15.423254  6500 solver.cpp:486] Iteration 612, lr = 0.0001
I0811 05:23:23.084491  6500 solver.cpp:214] Iteration 630, loss = -6.03199e-05
I0811 05:23:23.084550  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:23:23.084563  6500 solver.cpp:486] Iteration 630, lr = 0.0001
I0811 05:24:30.802199  6500 solver.cpp:214] Iteration 648, loss = -6.03199e-05
I0811 05:24:30.802258  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:24:30.802266  6500 solver.cpp:486] Iteration 648, lr = 0.0001
I0811 05:25:38.485126  6500 solver.cpp:214] Iteration 666, loss = 0.0311326
I0811 05:25:38.485186  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:25:38.485196  6500 solver.cpp:486] Iteration 666, lr = 0.0001
I0811 05:26:46.154042  6500 solver.cpp:214] Iteration 684, loss = 0.476817
I0811 05:26:46.154126  6500 solver.cpp:229]     Train net output #0: tripletloss = 0.774595 (\* 1 = 0.774595 loss)
I0811 05:26:46.154137  6500 solver.cpp:486] Iteration 684, lr = 0.0001
I0811 05:27:53.814285  6500 solver.cpp:214] Iteration 702, loss = 0.216171
I0811 05:27:53.814348  6500 solver.cpp:229]     Train net output #0: tripletloss = 1.01635 (\* 1 = 1.01635 loss)
I0811 05:27:53.814358  6500 solver.cpp:486] Iteration 702, lr = 0.0001
I0811 05:29:01.489145  6500 solver.cpp:214] Iteration 720, loss = 1.53251
I0811 05:29:01.489208  6500 solver.cpp:229]     Train net output #0: tripletloss = 0 (\* 1 = 0 loss)
I0811 05:29:01.489217  6500 solver.cpp:486] Iteration 720, lr = 0.0001
I0811 05:30:09.188515  6500 solver.cpp:214] Iteration 738, loss = 1.87498
I0811 05:30:09.188573  6500 solver.cpp:229]     Train net output #0: tripletloss = 0.0788164 (\* 1 = 0.0788164 loss)

**Comment by louielu1027** [comment]
为啥第一个loss有时正，有时负；第二个tripletloss总是一直为0，极少情况下才不为零？

**Comment by lliuz** [comment]
第一个Loss是多次取平均后的，第二个是当前iter的loss
